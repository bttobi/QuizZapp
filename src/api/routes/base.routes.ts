export const getBaseRoute = (): string => {
  return `http${import.meta.env.VITE_DOMAIN_SECURED ? 's' : ''}://${
    import.meta.env.VITE_SERVER_HOST
  }${
    import.meta.env.VITE_SERVER_PORT
      ? ':' + import.meta.env.VITE_SERVER_PORT
      : ''
  }`;
};
