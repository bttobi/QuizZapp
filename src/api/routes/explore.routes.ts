import { getBaseRoute } from '../routes/base.routes';

export const getExploreRoute = (page: number): string =>
  `${getBaseRoute()}/explore/${page}`;
