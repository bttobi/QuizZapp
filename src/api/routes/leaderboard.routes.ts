import { getBaseRoute } from './base.routes';

export const getLeaderboardRoute = (page: number): string =>
  `${getBaseRoute()}/leaderboard/${page}`;
