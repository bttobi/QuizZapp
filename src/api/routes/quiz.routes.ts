import { getBaseRoute } from './base.routes';

//TODO: do better route like in user routes
export const getQuizRoute = (quizID: number): string =>
  `${getBaseRoute()}/quiz/${quizID}`;

export const postAnswersRoute = (quizID: number): string =>
  `${getBaseRoute()}/answers/${quizID}`;

export const getResultsRoute = (quizID: number): string =>
  `${getBaseRoute()}/results/${quizID}`;

export const postCreateQuizRoute = (): string =>
  `${getBaseRoute()}/quiz/create`;

export const postDeleteQuizzesRoute = (): string =>
  `${getBaseRoute()}/quiz/delete`;

export const postEditQuizRoute = (quizID: number): string =>
  `${getBaseRoute()}/quiz/edit/${quizID}`;

export const getEditQuizRoute = (quizID: number): string =>
  `${getBaseRoute()}/quiz/edit/${quizID}`;

export const postCreateQuestionRoute = (quizID: number): string =>
  `${getBaseRoute()}/quiz/${quizID}/question/create`;

export const getQuestionsRoute = (quizID: number): string =>
  `${getBaseRoute()}/quiz/${quizID}/questions/edit`;

export const postEditQuestionRoute = (
  quizID: number,
  questionID: number
): string => `${getBaseRoute()}/quiz/${quizID}/question/edit/${questionID}`;

export const getEditQuestionRoute = (
  quizID: number,
  questionID: number
): string => `${getBaseRoute()}/quiz/${quizID}/question/edit/${questionID}`;

export const postDeleteQuestionsRoute = (quizID: number): string =>
  `${getBaseRoute()}/quiz/${quizID}/questions/delete`;
