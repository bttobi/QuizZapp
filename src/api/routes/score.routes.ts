import { getBaseRoute } from './base.routes';

export const getUserScoresRoute = (userID: number): string =>
  `${getBaseRoute()}/scores/all/${userID}`;

export const getStatsRoute = (): string => `${getBaseRoute()}/scores/all`;
