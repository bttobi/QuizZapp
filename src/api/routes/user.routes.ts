import { getBaseRoute } from './base.routes';

export const getUserRoute = (): string => `${getBaseRoute()}/user`;

export const signUpUserRoute = (): string => `${getUserRoute()}/signup`;

export const signInUserRoute = (): string => `${getUserRoute()}/signin`;

export const signOutUserRoute = (): string => `${getUserRoute()}/signout`;

export const refreshUserTokenRoute = (): string => `${getUserRoute()}/token`;

export const getUserRanking = (email: string): string =>
  `${getUserRoute()}/points/${email}`;
