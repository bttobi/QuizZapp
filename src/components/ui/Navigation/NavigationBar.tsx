import React from 'react';
import {
  Button,
  Navbar,
  NavbarBrand,
  NavbarContent,
  NavbarMenu,
  NavbarMenuToggle,
  useDisclosure,
} from '@nextui-org/react';
import { FaSignOutAlt } from 'react-icons/fa';
import QuizZappLogo from '../Logo/QuizZappLogo';
import NavigationLink from './NavigationLink';
import { RouteType } from '../../../appRoutes.routes';
import useUser from '../../../hooks/useUser';
import { NavLink } from 'react-router-dom';
import ProfileDropdown from './ProfileDropdown';
import messages from '../../../api/messages/messages.json';
import { useSignOutUser } from '../../../api/hooks/user.hooks';

interface NavigationBarProps {
  items: RouteType[];
}

const NavigationBar: React.FC<NavigationBarProps> = ({ items }) => {
  const { isOpen, onOpenChange } = useDisclosure();
  const { signOut } = useSignOutUser();
  const { isSignedIn } = useUser();

  return (
    <Navbar
      isMenuOpen={isOpen}
      onMenuOpenChange={onOpenChange}
      className="bottom-auto bg-backgroundSecondary w-full"
      maxWidth="full"
    >
      <NavbarContent>
        <NavbarMenuToggle
          aria-label={isOpen ? 'Close menu' : 'Open menu'}
          className="md:hidden"
        />
        <NavbarBrand>
          <NavLink
            className="flex justify-center align-center items-center hover:scale-105 ease-in-out duration-200"
            to="/"
          >
            <QuizZappLogo width={'35px'} height={'35px'} />
            <p className="font-bold text-xl sm:text-lg sm:visible hidden">
              QuizZapp
            </p>
          </NavLink>
        </NavbarBrand>
      </NavbarContent>
      <NavbarContent
        className="hidden md:flex vitems-center align-center gap-2 md:gap-8 md:mx-5"
        justify="center"
      >
        {items.map(el => (
          <NavigationLink key={`${el.path}-${el.name}`} path={el.path}>
            <span className="mr-2">{el.icon}</span>
            <span className="text-sm lg:text-lg">{el.name}</span>
          </NavigationLink>
        ))}
      </NavbarContent>
      <NavbarContent justify="end" className="w-min">
        {isSignedIn ? (
          <ProfileDropdown />
        ) : (
          <>
            <NavigationLink path="/signin">
              <Button color="secondary" variant="flat">
                {messages.signIn}
              </Button>
            </NavigationLink>
            <NavigationLink path="/signup">
              <Button color="secondary" variant="flat">
                {messages.signUp}
              </Button>
            </NavigationLink>
          </>
        )}
      </NavbarContent>
      <NavbarMenu className="bg-background/70 pt-8">
        {items.map(el => (
          <NavigationLink key={`${el.path}-${el.name}`} path={el.path}>
            {el.icon && <span className="text-white mr-2">{el.icon}</span>}
            <span onClick={() => onOpenChange()} className="text-white">
              {el.name}
            </span>
          </NavigationLink>
        ))}
        {isSignedIn && (
          <NavigationLink onClick={() => onOpenChange()} color="danger">
            <span onClick={() => onOpenChange()} className="mr-2">
              <FaSignOutAlt />
            </span>
            <span
              onClick={() => {
                onOpenChange();
                signOut();
              }}
            >
              Sign out
            </span>
          </NavigationLink>
        )}
      </NavbarMenu>
    </Navbar>
  );
};

export default NavigationBar;
