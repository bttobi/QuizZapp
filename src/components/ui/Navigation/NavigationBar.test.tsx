import React from 'react';
import NavigationBar from './NavigationBar';
import { render, waitFor } from '../../../tests/test-utils';
import routes from '../../../appRoutes.routes';

const setup = () =>
  render(<NavigationBar items={routes.filter(el => el.isNavVisible)} />);

describe('NavigationBar component', () => {
  it('renders the component', async () => {
    const { container } = setup();
    await waitFor(() => {
      expect(container).toBeInTheDocument();
    });
  });

  it('displays all the routes', async () => {
    const { getByText } = setup();
    await waitFor(() => {
      expect(getByText('Home')).toBeInTheDocument();
      expect(getByText('Explore')).toBeInTheDocument();
      expect(getByText('Create')).toBeInTheDocument();
      expect(getByText('Leaderboard')).toBeInTheDocument();
      expect(getByText('My quizzes')).toBeInTheDocument();
    });
  });

  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
