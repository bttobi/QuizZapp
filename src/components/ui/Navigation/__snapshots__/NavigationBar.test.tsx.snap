// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationBar component > matches snapshot 1`] = `
<div>
  <nav
    class="flex z-40 h-auto items-center justify-center data-[menu-open=true]:border-none sticky top-0 inset-x-0 backdrop-blur-lg data-[menu-open=true]:backdrop-blur-xl backdrop-saturate-150 bottom-auto bg-backgroundSecondary w-full"
    style="--navbar-height: 4rem;"
  >
    <header
      class="z-40 flex px-6 gap-4 w-full flex-row relative flex-nowrap items-center justify-between h-[var(--navbar-height)] max-w-full"
    >
      <ul
        class="flex gap-4 h-full flex-row flex-nowrap items-center data-[justify=start]:justify-start data-[justify=start]:flex-grow data-[justify=start]:basis-0 data-[justify=center]:justify-center data-[justify=end]:justify-end data-[justify=end]:flex-grow data-[justify=end]:basis-0"
        data-justify="start"
      >
        <button
          aria-label="Open menu"
          aria-pressed="false"
          class="group flex items-center justify-center w-6 h-full rounded-small tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 md:hidden"
          type="button"
        >
          <span
            class="sr-only"
          >
            open navigation menu
          </span>
          <span
            class="w-full h-full pointer-events-none flex flex-col items-center justify-center text-inherit group-data-[pressed=true]:opacity-70 transition-opacity before:content-[''] before:block before:h-px before:w-6 before:bg-current before:transition-transform before:duration-150 before:-translate-y-1 before:rotate-0 group-data-[open=true]:before:translate-y-px group-data-[open=true]:before:rotate-45 after:content-[''] after:block after:h-px after:w-6 after:bg-current after:transition-transform after:duration-150 after:translate-y-1 after:rotate-0 group-data-[open=true]:after:translate-y-0 group-data-[open=true]:after:-rotate-45"
          />
        </button>
        <div
          class="flex basis-0 flex-row flex-grow flex-nowrap justify-start bg-transparent items-center no-underline text-medium whitespace-nowrap box-border"
        >
          <a
            aria-current="page"
            class="flex justify-center align-center items-center hover:scale-105 ease-in-out duration-200 active"
            href="/"
          >
            <img
              class="mr-4 + undefined"
              height="35px"
              src="https://i.imgur.com/VRwVtXr.png"
              width="35px"
            />
            <p
              class="font-bold text-xl sm:text-lg sm:visible hidden"
            >
              QuizZapp
            </p>
          </a>
        </div>
      </ul>
      <ul
        class="h-full flex-row flex-nowrap items-center data-[justify=start]:justify-start data-[justify=start]:flex-grow data-[justify=start]:basis-0 data-[justify=center]:justify-center data-[justify=end]:justify-end data-[justify=end]:flex-grow data-[justify=end]:basis-0 hidden md:flex vitems-center align-center gap-2 md:gap-8 md:mx-5"
        data-justify="center"
      >
        <li
          class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold"
          data-active="true"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <div
              class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
              role="link"
              tabindex="0"
            >
              <span
                class="mr-2"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
                  />
                </svg>
              </span>
              <span
                class="text-sm lg:text-lg"
              >
                Home
              </span>
            </div>
          </a>
        </li>
        <li
          class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold"
        >
          <a
            class=""
            href="/explore"
          >
            <div
              class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-foreground no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
              role="link"
              tabindex="0"
            >
              <span
                class="mr-2"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 10.9c-.61 0-1.1.49-1.1 1.1s.49 1.1 1.1 1.1c.61 0 1.1-.49 1.1-1.1s-.49-1.1-1.1-1.1zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm2.19 12.19L6 18l3.81-8.19L18 6l-3.81 8.19z"
                  />
                </svg>
              </span>
              <span
                class="text-sm lg:text-lg"
              >
                Explore
              </span>
            </div>
          </a>
        </li>
        <li
          class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold"
        >
          <a
            class=""
            href="/create"
          >
            <div
              class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-foreground no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
              role="link"
              tabindex="0"
            >
              <span
                class="mr-2"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
                  />
                </svg>
              </span>
              <span
                class="text-sm lg:text-lg"
              >
                Create
              </span>
            </div>
          </a>
        </li>
        <li
          class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold"
        >
          <a
            class=""
            href="/leaderboard"
          >
            <div
              class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-foreground no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
              role="link"
              tabindex="0"
            >
              <span
                class="mr-2"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M552 64H448V24c0-13.3-10.7-24-24-24H152c-13.3 0-24 10.7-24 24v40H24C10.7 64 0 74.7 0 88v56c0 35.7 22.5 72.4 61.9 100.7 31.5 22.7 69.8 37.1 110 41.7C203.3 338.5 240 360 240 360v72h-48c-35.3 0-64 20.7-64 56v12c0 6.6 5.4 12 12 12h296c6.6 0 12-5.4 12-12v-12c0-35.3-28.7-56-64-56h-48v-72s36.7-21.5 68.1-73.6c40.3-4.6 78.6-19 110-41.7 39.3-28.3 61.9-65 61.9-100.7V88c0-13.3-10.7-24-24-24zM99.3 192.8C74.9 175.2 64 155.6 64 144v-16h64.2c1 32.6 5.8 61.2 12.8 86.2-15.1-5.2-29.2-12.4-41.7-21.4zM512 144c0 16.1-17.7 36.1-35.3 48.8-12.5 9-26.7 16.2-41.8 21.4 7-25 11.8-53.6 12.8-86.2H512v16z"
                  />
                </svg>
              </span>
              <span
                class="text-sm lg:text-lg"
              >
                Leaderboard
              </span>
            </div>
          </a>
        </li>
        <li
          class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold"
        >
          <a
            class=""
            href="/quizzes"
          >
            <div
              class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-foreground no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
              role="link"
              tabindex="0"
            >
              <span
                class="mr-2"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6z"
                  />
                  <path
                    d="M20 2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-5.99 13c-.59 0-1.05-.47-1.05-1.05 0-.59.47-1.04 1.05-1.04.59 0 1.04.45 1.04 1.04-.01.58-.45 1.05-1.04 1.05zm2.5-6.17c-.63.93-1.23 1.21-1.56 1.81-.13.24-.18.4-.18 1.18h-1.52c0-.41-.06-1.08.26-1.65.41-.73 1.18-1.16 1.63-1.8.48-.68.21-1.94-1.14-1.94-.88 0-1.32.67-1.5 1.23l-1.37-.57C11.51 5.96 12.52 5 13.99 5c1.23 0 2.08.56 2.51 1.26.37.61.58 1.73.01 2.57z"
                  />
                </svg>
              </span>
              <span
                class="text-sm lg:text-lg"
              >
                My quizzes
              </span>
            </div>
          </a>
        </li>
      </ul>
      <ul
        class="flex gap-4 h-full flex-row flex-nowrap items-center data-[justify=start]:justify-start data-[justify=start]:flex-grow data-[justify=start]:basis-0 data-[justify=center]:justify-center data-[justify=end]:justify-end data-[justify=end]:flex-grow data-[justify=end]:basis-0 w-min"
        data-justify="end"
      >
        <li
          class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold"
        >
          <a
            class=""
            href="/signin"
          >
            <div
              class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-foreground no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
              role="link"
              tabindex="0"
            >
              <button
                class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-unit-4 min-w-unit-20 h-unit-10 text-small gap-unit-2 rounded-medium [&>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-secondary/20 text-secondary data-[hover=true]:opacity-hover"
                type="button"
              >
                Sign in
              </button>
            </div>
          </a>
        </li>
        <li
          class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold"
        >
          <a
            class=""
            href="/signup"
          >
            <div
              class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-foreground no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
              role="link"
              tabindex="0"
            >
              <button
                class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-unit-4 min-w-unit-20 h-unit-10 text-small gap-unit-2 rounded-medium [&>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-secondary/20 text-secondary data-[hover=true]:opacity-hover"
                type="button"
              >
                Sign up
              </button>
            </div>
          </a>
        </li>
      </ul>
    </header>
  </nav>
  <div
    class="Toastify"
  />
</div>
`;
